### PR TITLE
Streamline reading of int TestNG props from Mvn

### LINKED ISSUE #574: String cannot be cast to Integer if log property is set in maven pom.xml
Hi,  
I am using maven to run the TestNG tests. For this I created some suite.xml files like:  

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
<suite name="Suite Name" verbose="10" parallel="classes">
    <test name="Web Channel Tests for B2B">
        <packages>
            <package name="com.some.package" />
        </packages>
    </test>
</suite>
```

There I set the verbosity to 10, but the generated verbosity after running the test is 0.  
So I thought, I could set the verbosity from maven to 10. This is possible by setting the properties parameter 

``` xml
<build>
    <plugins>
        <plugin>
            <artifactId>maven-surefire-plugin</artifactId>
            <version>2.16</version>
            <configuration>
                <properties>
                    <property>
                        <name>log</name>
                        <value>10</value>
                    </property>
                </properties>
                <suiteXmlFiles>
                    <suiteXmlFile>Some_suite.xml</suiteXmlFile>
                </suiteXmlFiles>
            </configuration>
        </plugin>
    </plugins>
</build>
```

Unfortunately TestNG dumps when reading this property, because the code in TestNG 6.8.7 looks like this:

``` java
   Integer verbose = (Integer) cmdLineArgs.get(CommandLineArgs.LOG);
    if (null != verbose) {
      result.verbose = verbose;
    }
```

The cast seems to be wrong there. The code for the threadcount setting (which can also be set by a maven property) has a conversion from String to int:

``` java
    String threadCount = (String) cmdLineArgs.get(CommandLineArgs.THREAD_COUNT);
    if (threadCount != null) {
      result.threadCount = Integer.parseInt(threadCount);
    }
```

Could you please fix the conversion of the log parameter or the generation of the correct verbosity from the initial suite.xml files?

My used versions:  
- maven-surefire-plugin 2.18
- TestNG 6.8.7

Thanks,

Iwan

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #574

Fixes #574 .

### Did you remember to?

- [ ] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/main/java/org/testng/TestNG.java [hunk 1]
```diff
@@ -1530,14 +1530,9 @@ public void configure(Map cmdLineArgs) {
       result.configFailurePolicy = failurePolicy;
     }
```
Comment (juherr) [inline src/main/java/org/testng/TestNG.java]: It won't work with Integer which may be provided by 3rd party integration.

**Review by juherr — CHANGES_REQUESTED** [review]
I'm not sure it will fix the issue and I feel it will open new ones.

File: src/main/java/org/testng/TestNG.java (hunk 1 above)
Comment (krmahadevan) [reply src/main/java/org/testng/TestNG.java]: I didn't quite understand what that means. I have tried this locally with Maven and it works. All we are doing here is to accept the parameter as a String and then parse it. Can you please tell me in which circumstances and with which 3rd party integrations it wouldn't work ?

Comment (juherr) [reply src/main/java/org/testng/TestNG.java]: Did you test old surefire versions? Does it work with the command line or ant integration?
What about IDE plugins?
Just be sure the Integer version is not used somewhere.

Comment (krmahadevan) [reply src/main/java/org/testng/TestNG.java]: @juherr - I have only tested this with surefire plugin.  So going by what you are saying there can be just two possibilities. 

1. The 3rd party integration provided us with an `Integer` (And we attempted at parsing it as `String` thus causing a `ClassCastException`
2. The 3rd party integration provided us with a `String` and our implementation works.

I guess I will go back and change this to run an `instanceof` check and do it accordingly. 

Have amended my commit and pushed in the changes for this. Please help take a look.

**Review by juherr — APPROVED** [review]
Ping @cbeust

**Pull request merged**